packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneAndTalkButton/DoneAndTalkButton.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Button, Text } from 'grommet'
+import styled, { ThemeProvider } from 'styled-components'
+import theme from 'styled-theming'
+import { darken, lighten } from 'polished'
+import zooTheme from '@zooniverse/grommet-theme'
+import counterpart from 'counterpart'
+import en from './locales/en'
+
+counterpart.registerTranslations('en', en)
+
+// Taken from PFE on 2019.02.28; these colours aren't part of the current theme.
+const TALK_LINK_BLUE = '#43bbfd';
+const TALK_LINK_BLUE_HOVER = darken(0.18, TALK_LINK_BLUE);
+const TALK_LINK_BLUE_HOVER_DARK = '#104A79';
+
+// TODO move what makes sense into theme
+export const StyledDoneAndTalkButton = styled(Button)`
+  background-color: ${theme('mode', {
+    dark: zooTheme.global.colors['dark-1'],
+    light: TALK_LINK_BLUE
+  })};
+  border: ${theme('mode', {
+    dark: `solid thin ${TALK_LINK_BLUE}`,
+    light: `solid thin ${TALK_LINK_BLUE_HOVER}`
+  })};
+  box-shadow: none;
+  color: white;
+  flex: 3 0;
+  margin-right: 1ch;
+  padding: 0.5em;
+  text-transform: capitalize;
+  
+  > i {
+    margin-left: 1ch;
+  }
+
+  &:hover, &:focus {
+    background: ${theme('mode', {
+      dark: TALK_LINK_BLUE_HOVER_DARK,
+      light: TALK_LINK_BLUE_HOVER
+    })};
+    border: ${theme('mode', {
+      dark: `solid thin ${TALK_LINK_BLUE}`,
+      light: `solid thin ${TALK_LINK_BLUE_HOVER}`
+    })};
+    box-shadow: none;
+    color: ${theme('mode', {
+      dark: zooTheme.global.colors.text.dark,
+      light: `white`
+    })};
+  }
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+  `
+
+function openTalkLinkAndClick (event, props) {
+  props.onClick(event)
```

Comment:
Should this happen after the new window opens, for safety, since submitting a classification changes the current subject?

Reply:
I've refactored the onSubmit handler and the classification store action to return the promise of POSTing the classification. The handling of what to load in the window is now in the promise's then.

Reply:
I'll take a look but shouldn't the new window open before the promise resolves, to avoid the old PFE bug? https://github.com/zooniverse/Panoptes-Front-End/issues/3679

Reply:
I don't see the old PFE buggy behavior with this. I'm testing manually and preserving the network tab log and can see the subject linked to the POSTed classification is the same id as what's used in the URL.

Reply:
👍 I never really understood that PFE bug, but it might have been something to do with the `a` tag's href attribute changing while the link was opening (which is a massive XSS vector if true.)

Code here looks good to me. Now that it's getting `props` from the outer function scope, that may make it safer. At least, my reading of the new code is that the value of `talkURL` won't be changed by the click event, because it's set on initial render.